### DISCUSSION TITLE
Is it possible to remove drives from the "Visited" map

### DISCUSSION THREAD

**Discussion by burratha** [teslamate-org/teslamate#3326] [General]
I recently had a breakdown (and recovery) across country which resulted in a bit of a mess in the "Visited" map, resulting in a single long straight line between two points.

I've since removed the drives from the date in question from the database, but the map still shows a route/entries for that date.Is it possible to get the platform to redraw the map from the entries, or even remove the drives from the map itself?

In summary: I have no drives listed for a particular date, but the "visited" map still shows the route(s) taken on that day.

**Comment by cwanja** [comment]
Remove the data from the positions table. However be aware that it could still produce a straight line. 

Take a backup of your data before such that if you are unhappy you can restore the data back.

**Comment by chrislombardo** [comment]
I have a very similar issue.  I deleted the drive connected with the error in the Visited map, but the incorrect position still exists.  Could someone please detail how to delete an entry from the positions table?  I feel like this would be a great addition to https://docs.teslamate.org/docs/maintenance/manually_fixing_data.

**Reply by cwanja** [reply]
Do you recall the drive ID you removed? Here is what I have provided in the past - two options: knowing the drive ID or figuring out the positions ID from the timestamps. 

Option 1 - known drive ID
> `delete from positions where drive_id = ###;` where ### is the ID from the drive ID in the URL.

Option 2 - figuring out the position IDs
>`select id, date, latitude, longitude from positions order by id asc;`
> 
> Identify the positions row IDs that are causing the out of band data based upon the timestamps against the drive details dashboard or by plugging the longitude / latitude into your preferred map provider. Then run the following delete command:
>
> `delete from positions where CAST(id as integer) between 1234 and 9876;` where 1234 is the starting point of positions records you want deleted and 9876 is the ending point.

As always, backup your data before running any of these.

**Reply by chrislombardo** [reply]
Perfect!  I did have the drive id, so was able to remove the erroneous position.  Thank you so much!

**Comment by burratha** [comment]
I have no idea which “drive” caused the issues; the car had a puncture in
another county and i had to have it recovered home.

I deleted all drives for that day, but the map entry still exists.

Any idea how I can just remove all “visited” entries for that day?

Steve

**Reply by cwanja** [reply]
Use the `select` commands from above and add a `where` clause filtering on the date to get the position IDs from that day and then run the `delete` command for the starting and ending position ID. 

Alternatively, if you know the date, use that to find the missing drive IDs in the database and then plug those drive IDs into the `delete` command for option one.

**Comment by burratha** [comment]
Thank you. Might have to wait until next week as I’m away at the moment and
the pi is on my lan and not accessible from outside.

Much appreciated.

Steve

**Comment by burratha** [comment]
Is it possible to run this command between two dates? I know the date that
the glitch occured, but it's 8 months from the beginning of my data.

Thanks

**Reply by cwanja** [reply]
You could structure an `or` clause in the `where` statement to add a layer of filtering on different dates. https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-where/

**Comment by burratha** [comment]
Sorry – I’m not sure I understand. I was thinking something along the lines of… select id, date, latitude, longitudefrom positionswhere date >= 2023-09-01 00:00:01order by id desc;   From: cwanja ***@***.***>Date: Wednesday, 27 September 2023 at 18:58To: adriankumpf/teslamate ***@***.***>Cc: burratha ***@***.***>, Author ***@***.***>Subject: Re: [adriankumpf/teslamate] Is it possible to remove drives from the "Visited" map (Discussion #3326)You could structure an or clause in the where statement to add a layer of filtering on different dates. https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-where/—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Reply by cwanja** [reply]
Sure that could also work, you would just get every possible result greater than and equal to that date. I assumed you were trying to _only_ get the results for the two dates. 

PostgreSQL is just like any other SQL query language.

**Comment by burratha** [comment]
Tbh, I have no idea what I’m doing… haha I need to remove all drives that occurred on 22 and 23 August 2023so…select id, date, latitude, longitudefrom positionswhere date >= 2023-08-22 00:00:01 AND date <= 2023-08-23 23:59:59order by id desc; Would that work? From: cwanja ***@***.***>Date: Wednesday, 27 September 2023 at 19:37To: adriankumpf/teslamate ***@***.***>Cc: burratha ***@***.***>, Author ***@***.***>Subject: Re: [adriankumpf/teslamate] Is it possible to remove drives from the "Visited" map (Discussion #3326)Sure that could also work, you would just get every possible result greater than and equal to that date. I assumed you were trying to only get the results for the two dates.PostgreSQL is just like any other SQL query language.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by burratha** [comment]
Hmmm maybe not  teslamate=# select id, date, latitude, longitudefrom positionswhere date >= 2023-08-22 00:00:01 AND date <= 2023-08-23 23:59:59order by id desc;select id, date, latitude, longitudefrom positionswhere date >= 2023-08-22 00:00:01 AND date <= 2023-08-23 23:59:59order by id desc;ERROR:  syntax error at or near "00"LINE 3: where date >= 2023-08-22 00:00:01 AND date <= 2023-08-23 23:...                                 ^ERROR:  syntax error at or near "00"LINE 3: where date >= 2023-08-22 00:00:01 AND date <= 2023-08-23 23:...               From: ***@***.*** ***@***.***>Date: Wednesday, 27 September 2023 at 19:42To: adriankumpf/teslamate ***@***.***>, adriankumpf/teslamate ***@***.***>Cc: Author ***@***.***>Subject: Re: [adriankumpf/teslamate] Is it possible to remove drives from the "Visited" map (Discussion #3326)Tbh, I have no idea what I’m doing… haha I need to remove all drives that occurred on 22 and 23 August 2023so…select id, date, latitude, longitudefrom positionswhere date >= 2023-08-22 00:00:01 AND date <= 2023-08-23 23:59:59order by id desc; Would that work? From: cwanja ***@***.***>Date: Wednesday, 27 September 2023 at 19:37To: adriankumpf/teslamate ***@***.***>Cc: burratha ***@***.***>, Author ***@***.***>Subject: Re: [adriankumpf/teslamate] Is it possible to remove drives from the "Visited" map (Discussion #3326)Sure that could also work, you would just get every possible result greater than and equal to that date. I assumed you were trying to only get the results for the two dates.PostgreSQL is just like any other SQL query language.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Reply by cwanja** [reply]
`Select` is only going to display the values, but a good first step to verify you are pulling and filtering to the right values you want.

I was able make a select statement for values only on 2023-09-02 by running:
`select id, date from positions where date >= '2023-09-02' AND date <= '2023-09-03';`

Once you have the values, you would then run the `delete` command. My example uses the `ID` as the delete value identifier, you would need to change it to `date`. 

Also, FYI - your replies are broken up into individual responses on the GitHub discussion making tracking challenging. And the most recent replies include CSS styling.

**Reply by burratha** [reply]
Sorry. I was responding to the emails, rather than posting on the site. The last response was using outlook.

**Reply by burratha** [reply]
FINALLY... managed to have a real go at this and your explanation and instructions have worked wonders. Thank you so much, cwanja.

**Reply by cwanja** [reply]
Glad it worked!
